### PR TITLE
Some small fixes in opmapcontrol

### PR DESCRIPTION
**Pull request by hdino**
While using the opmapcontrol lib in another project I came across some issues like memory leaks and deprecated project files. The committed changes fix those.

### CODE REVIEW

File: libs/opmapcontrol/src/internals/core.cpp [hunk 1]
```diff
@@ -65,6 +66,9 @@ namespace internals {
     Core::~Core()
     {
         ProcessLoadTaskCallback.waitForDone();
+        Matrix.Clear();
+        if (projection)
```
Comment (Arne-W) [inline libs/opmapcontrol/src/internals/core.cpp]: deleting a null-pointer is a nop. So I would prefer something like this:
`delete projection;
projection = 0;`
Another possibility would be using an autoPointer.

File: libs/opmapcontrol/src/internals/core.h [hunk 2]
```diff
@@ -135,6 +135,9 @@ namespace internals {
         }
         void SetProjection(PureProjection* value)
         {
+            if (projection)
+                delete projection;
```
Comment (Arne-W) [inline libs/opmapcontrol/src/internals/core.h]: See above

File: libs/opmapcontrol/src/internals/pureprojection.h [hunk 3]
```diff
@@ -44,6 +44,8 @@ class PureProjection
 
 
 public:
+    virtual ~PureProjection() {}
```
Comment (Arne-W) [inline libs/opmapcontrol/src/internals/pureprojection.h]: :+1:  very good finding!

**Comment by hdino** [comment]
Thanks for your review! I didn't want to use an autopointer since the whole lib uses raw pointers. I think I would be better to change all pointers at once to keep the code consistent.

**Comment by Arne-W** [comment]
Thanks for the update - and yes only switching one pointer to an automatic one could be confusing.

**Pull request merged**